src/main/kotlin/net/lumalyte/lg/LumaGuilds.kt

```
@@ -961,6 +969,38 @@ class LumaGuilds : JavaPlugin() {
+        try {
+            val relationService = get().get<net.lumalyte.lg.application.services.RelationService>()
+            val intervalTicks = 20L * 60L * 5L // every 5 minutes
+            val initialDelayTicks = 20L * 60L  // first run after 1 minute
+            server.scheduler.runTaskTimer(this, Runnable {
+                try {
+                    val expired = relationService.processExpiredRelations()
+                    val cleaned = relationService.cleanupStaleRelations()
+                    if (expired > 0 || cleaned > 0) {
+                        logger.info("Relation maintenance: expired=$expired, cleaned=$cleaned")
+                    }
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)